example/core_tutorial/05_slice/slice_example.cpp

```diff
@@ -56,16 +56,15 @@ void sliceExample()
     const int VectorLength = 4;
 
     /*
-      Finally declare the memory space in which the AoSoA will be allocated. In
-      this example we are writing basic loops that will execute on the CPU. The
-      HostSpace allocates memory in standard CPU RAM.
+      The current example uses default spaces to work across all supported
+      backends, but explicit choices like HostSpace and CudaSpace can also be
+      used.
 
-      Kokkos also supports execution on GPUs. For example, to create an AoSoA
-      allocated on NVIDIA devices use `Kokkos::CudaSpace` instead of
-      `Kokkos::HostSpace`.
+      We declare the memory space in which the AoSoA will be allocated
+      on the default execution space we are dealing with.
     */
-    using MemorySpace = Kokkos::HostSpace;
-    using ExecutionSpace = Kokkos::DefaultHostExecutionSpace;
+    using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+    using MemorySpace = typename ExecutionSpace::memory_space;
     using DeviceType = Kokkos::Device<ExecutionSpace, MemorySpace>;
```

Comment:
Just a note that #585 will change most of the examples to use `memory_space` directly - it may be best to wait until that is merged

Reply:
Sure, I will update the rest of the examples after #585 is merged. Will keep the PR open until then.

Reply:
@dineshadepu should be okay to continue this PR now that we've merged the blocking PR - let me know if those changes are confusing in any way. But I do not plan to include this in the imminent release, so feel free to wait until that's finished